cpp/cmake/BuildUtils.cmake

```diff
@@ -83,30 +83,42 @@ function(build_external PROJ)
                    EXCLUDE_FROM_ALL)
 endfunction(build_external PROJ)
 
-# Add "make lint" target
-function(add_lint)
+find_program(CLANG_FORMAT_BIN NAMES clang-format)
+message(STATUS "Found clang-format executable at ${CLANG_FORMAT_BIN}")
+
+file(GLOB_RECURSE LINT_FILES
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/*.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/src/*.cc"
+  "${CMAKE_CURRENT_SOURCE_DIR}/integ/*.h"
+  "${CMAKE_CURRENT_SOURCE_DIR}/integ/*.cc"
```

Comment:
need to add includes too (or) is it pulled from its references?

Reply:
i missed that - fixed it now.